runtime/compiler/runtime/ClassUnloadAssumption.cpp

```diff
@@ -879,31 +857,34 @@ TR_RuntimeAssumptionTable::notifyClassRedefinitionEvent(TR_FrontEnd *vm, bool is
    raArray = findAssumptionHashTable(RuntimeAssumptionOnClassRedefinitionNOP)->_htSpineArray;
    if (reportDetails)
       {
-      TR_VerboseLog::vlogAcquire();
+      TR_VerboseLog::CriticalSection vlogLock;
       TR_VerboseLog::writeLine(TR_Vlog_RA,"Scanning for NOP assumptions for %p in array %p bucket %p", oldKey, raArray, oldHeadPtr);
       if (!nop_cursor)
          TR_VerboseLog::writeLine(TR_Vlog_RA,"oldKey %p not registered with NOP!", oldKey);
-      TR_VerboseLog::vlogRelease();
       }
 
    while (nop_cursor)
       {
       OMR::RuntimeAssumption* nop_next = nop_cursor->getNext();
+
+      TR_VerboseLog::CriticalSection vlogLock(reportDetails);
       if (reportDetails)
-         TR_VerboseLog::writeLine(TR_Vlog_RA, "old=%p @ %p", nop_cursor->getKey(), nop_cursor->getFirstAssumingPC());
+         TR_VerboseLog::write(TR_Vlog_RA, "old=%p @ %p", nop_cursor->getKey(), nop_cursor->getFirstAssumingPC());
+
       if (nop_cursor->matches((uintptr_t)oldKey))
          {
          if (reportDetails)
-            {
-            TR_VerboseLog::vlogAcquire();
-            TR_VerboseLog::write(" compensating new=%p", newKey);
-            TR_VerboseLog::vlogRelease();
-            }
+            TR_VerboseLog::writeLine(" compensating new=%p", newKey);
+
          nop_cursor->compensate(vm, 0, 0);
          markForDetachFromRAT(nop_cursor);
          nop_cursor = nop_next;
          continue;
          }
+
+      if (reportDetails)
+         TR_VerboseLog::writeLine("");
```

Comment:
I don't think we need this; it would just add an empty new line. I _think_ the reason the earlier code used to do `writeLine` first and then `write` is because before the vlog writing code was fixed, the new line was added to the start of the line. 

If the concern is that we're not always going to print `compensating new=...` and thus not print a new line, then this should be added within the else clause of `if (nop_cursor->matches((uintptr_t)oldKey))`. Either that, or `TR_VerboseLog::writeLine(" compensating new=%p", newKey);` should be changed to `TR_VerboseLog::write(" compensating new=%p", newKey);`

Reply:
> If the concern is that we're not always going to print `compensating new=...` and thus not print a new line, then this should be added within the else clause of `if (nop_cursor->matches((uintptr_t)oldKey))`.

It's already impossible for the `if`-body to fall through to here because of the `continue` a few lines up

Reply:
Oh I see, I missed the `continue`.